azure/scope/machinepool.go

```diff
@@ -72,6 +74,15 @@ type (
 		patchHelper                *patch.Helper
 		capiMachinePoolPatchHelper *patch.Helper
 		vmssState                  *azure.VMSS
+		cache                      *MachinePoolCache
+	}
+
+	// MachinePoolCache stores common machine pool information so we don't have to hit the API multiple times within the same reconcile loop.
+	MachinePoolCache struct {
+		BootstrapData string
```

Comment:
are we using `BootstrapData`, `VMImage` and `VMSKU` from the cache anywhere in this PR? If not let's omit them and only add `VMSSKU` for now. @Jont828 is adding the rest in #2719

Reply:
Nope they are not, I'll go ahead and remove them!